README.md

```diff
@@ -22,21 +22,529 @@ A Spezi framework that provides a common set of SwiftUI views and related functi
+                    Divider()
+                        .gridCellUnsizedAxes(.horizontal)
+                    
+                    NameFieldRow("Middle", name: $name, for: \.middleName) {
+                        Text("Enter your middle name (optional)")
+                    }
+                    
+                    Divider()
+                        .gridCellUnsizedAxes(.horizontal)
+                    
+                    NameFieldRow("Last", name: $name, for: \.familyName) {
+                        Text("Enter your last name")
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+### SpeziValidation - Input Validation
+
+Add real-time validation to forms:
+
+```swift
+import SpeziValidation
+import SwiftUI
+
+struct SignupView: View {
+    @State private var email = ""
+    @State private var password = ""
+    @State private var username = ""
+    
+    @ValidationState private var validation
+    
+    var body: some View {
+        Form {
+            Section("Account Details") {
+                VerifiableTextField("Email", text: $email)
+                    .validate(input: email, rules: .minimalEmail)
+                    .textInputAutocapitalization(.never)
+                    .autocorrectionDisabled()
+                
+                VerifiableTextField("Password", text: $password, type: .secure)
+                    .validate(input: password, rules: .minimalPassword)
+                
+                VerifiableTextField("Username", text: $username)
+                    .validate(input: username, rules: .nonEmpty)
+            } footer: {
+                Text("Choose a unique username")
+            }
+        }
+        .receiveValidation(in: $validation)
+        .toolbar {
+            ToolbarItem(placement: .primaryAction) {
+                Button("Create Account") {
+                    guard validation.validateSubviews() else {
+                        return
+                    }
+                    // Create account
+                }
+                .disabled(!validation.allInputValid)
+            }
+        }
+    }
+}
+```
+
+#### Custom Validation Rules
+
+Create custom validation logic:
+
+```swift
+import SpeziValidation
+import SwiftUI
+
+extension ValidationRule {
+    static let strongPassword = ValidationRule(
+        rule: { password in
+            password.count >= 8 && 
+            password.contains { $0.isUppercase } &&
+            password.contains { $0.isLowercase } &&
+            password.contains { $0.isNumber }
+        },
+        message: "Password must be at least 8 characters with uppercase, lowercase, and numbers"
+    )
+}
+
+struct SecureSignupView: View {
+    @State private var password = ""
+    
+    var body: some View {
+        Form {
+            VerifiableTextField("Password", text: $password, type: .secure)
+                .validate(input: password, rules: .strongPassword)
+        }
+    }
+}
+```
+
+#### Validation with Multiple Rules
```

Comment:
I think the detailed elements might rather move to the main DocC document of the target and ideally with a bit more explanations and context + motivation. I think there are some great examples in the inline docs and we might want to use them and/or link to some explanations here. That connectivity can help users (and hopefully AI as well) to connect the pieces of the documentation.